### PR TITLE
LPS-162472 LPS-160493 cause about -8.44% performance regression

### PR DESCRIPTION
**Pull request by cgoncas**
### Motivation

Fix [LPS-162472 LPS-160493 cause about -8.44% performance regression](https://issues.liferay.com/browse/LPS-162472).

Now we are calling the AC endpoint to calculate the value isAnalyticsSync in every page render, because we render the button: com.liferay.segments.experiment.web.internal.product.navigation.control.menu.SegmentsExperimentProductNavigationControlMenuEntry

### Proposed solution

Move the code from the com.liferay.segments.experiment.web.internal.display.context.SegmentsExperimentDisplayContext to a new Endpoint /segments_experiment/get_data, so in the SegmentsExperimentProductNavigationControlMenuEntry, we only fill the value "getDataEndpoint" that point to /segments_experiment/get_data

1. Add new field in SegmentsExperimentDisplayContext that returns the dataURL:
`"segmentExperimentDataURL":  http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=2&p_p_state=maximized&p_p_resource_id=%2Fsegments_experiment%2Fget_data&p_p_cacheability=cacheLevelPage&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_redirect=http%3A%2F%2Flocalhost%3A8080%2Faba41e86-0e10-bfd1-7055-7574770ea41a%3Fp_l_back_url%3Dhttp%253A%252F%252Flocalhost%253A8080%252Fhome%253FsegmentsExperienceId%253D43030%26p_l_mode%3Dedit%26redirect%3Dhttp%253A%252F%252Flocalhost%253A8080%252Faba41e86-0e10-bfd1-7055-7574770ea41a%253Fp_l_back_url%253Dhttp%25253A%25252F%25252Flocalhost%25253A8080%25252Fhome%25253FsegmentsExperienceId%25253D43030%2526p_l_mode%253Dedit&p_l_back_url=http%3A%2F%2Flocalhost%3A8080%2Fhome%3FsegmentsExperienceId%3D43030&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_plid=6&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_segmentsExperienceId=43030`
2. Create new resource MVC Command GetDataMVCResourceCommand that returns all the information for the component:  

```{
  "context": {
    "endpoints": {
      "runSegmentsExperimentURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Frun_segments_experiment&p_l_mode=view&p_auth=eufybh4n",
      "createSegmentsVariantURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_layout_content_page_editor_web_internal_portlet_ContentPageEditorPortlet&p_p_lifecycle=1&_com_liferay_layout_content_page_editor_web_internal_portlet_ContentPageEditorPortlet_javax.portlet.action=%2Flayout_content_page_editor%2Fadd_segments_experience&p_auth=eufybh4n&p_p_auth=RkN47QEG&p_l_mode=edit",
      "deleteSegmentsExperimentURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fdelete_segments_experiment&p_l_mode=view&p_auth=eufybh4n",
      "editSegmentsVariantURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fedit_segments_experiment_rel&p_l_mode=view&p_auth=eufybh4n",
      "createSegmentsExperimentURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fadd_segments_experiment&p_l_mode=view&p_auth=eufybh4n",
      "editSegmentsVariantLayoutURL": "http://localhost:8080/aba41e86-0e10-bfd1-7055-7574770ea41a?p_l_back_url=&p_l_mode=edit&redirect=http%3A%2F%2Flocalhost%3A8080%2Faba41e86-0e10-bfd1-7055-7574770ea41a%3Fredirect%3Dhttp%253A%252F%252Flocalhost%253A8080%252Faba41e86-0e10-bfd1-7055-7574770ea41a%253Fp_l_back_url%253Dhttp%25253A%25252F%25252Flocalhost%25253A8080%25252Fhome%25253FsegmentsExperienceId%25253D43030%2526p_l_mode%253Dedit%26p_l_back_url%3D%26p_l_mode%3Dedit",
      "editSegmentsExperimentURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fedit_segments_experiment&p_l_mode=view&p_auth=eufybh4n",
      "editSegmentsExperimentStatusURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fedit_segments_experiment_status&p_l_mode=view&p_auth=eufybh4n",
      "deleteSegmentsVariantURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fdelete_segments_experiment_rel&p_l_mode=view&p_auth=eufybh4n",
      "calculateSegmentsExperimentEstimatedDurationURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fcalculate_segments_experiment_estimated_duration&p_l_mode=view&p_auth=eufybh4n"
    },
    "contentPageEditorNamespace": "_com_liferay_layout_content_page_editor_web_internal_portlet_ContentPageEditorPortlet_",
    "imagesPath": "/o/segments-experiment-web/images",
    "namespace": "_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_",
    "page": {
      "classPK": "6",
      "classNameId": 20032,
      "type": "content"
    }
  },
  "props": {
    "historySegmentsExperiments": [],
    "analyticsData": {
      "isSynced": false,
      "isConnected": false,
      "cloudTrialURL": "https://www.liferay.com/products/analytics-cloud/get-started"
    },
    "initialSegmentsVariants": [],
    "winnerSegmentsVariantId": "",
    "pathToAssets": "/o/segments-experiment-web",
    "segmentsExperimentGoals": [
      {
        "label": "Bounce Rate",
        "value": "bounce-rate"
      },
      {
        "label": "Click",
        "value": "click"
      }
    ],
    "hideSegmentsExperimentPanelURL": "http://localhost:8080/group/guest/~/control_panel/manage?p_p_id=com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet&p_p_lifecycle=1&p_p_state=maximized&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_javax.portlet.action=%2Fsegments_experiment%2Fhide_segments_experiment_panel&_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_redirect=http%3A%2F%2Flocalhost%3A8080%2Faba41e86-0e10-bfd1-7055-7574770ea41a%3Fp_l_back_url%3Dhttp%253A%252F%252Flocalhost%253A8080%252Fhome%253FsegmentsExperienceId%253D43030%26p_l_mode%3Dedit%26redirect%3Dhttp%253A%252F%252Flocalhost%253A8080%252Faba41e86-0e10-bfd1-7055-7574770ea41a%253Fp_l_back_url%253Dhttp%25253A%25252F%25252Flocalhost%25253A8080%25252Fhome%25253FsegmentsExperienceId%25253D43030%2526p_l_mode%253Dedit&p_l_mode=view&p_auth=eufybh4n",
    "selectedSegmentsExperienceId": "43030",
    "segmentsExperiences": [
      {
        "segmentsExperienceId": "43030",
        "name": "Default"
      }
    ]
  }
}
```

### Frontend perspective

From the frontend perspective the major changes are:

- We load the JS files and React components in the initial page load, but the API calls for fetching the data are saved and only executed on user interaction.
- The initial data the main React component receives is
```
{
  "isPanelStateOpen": false,
  "segmentExperimentDataURL": "url-for-fetch-the-data",
  "namespace": "_com_liferay_segments_experiment_web_internal_portlet_SegmentsExperimentPortlet_"
}
```
- Once the user clicks or focuses the topbar button, the request is sent and the rest of the data is fetched.
- We keep track of the state of the panel using `session_clicks`, so we know if the fetch needs to be done, in case the panel is initially open.
- The main component `SegmentsExperimentApp.es.js` is now split with another one `SegmentsExperimentsMain.es.js`, serving the first as a wrapper to prevent the actions be executed on first load. This way is easier to manage.

### Note
It would be important to approach the current https://github.com/liferay/liferay-portal/blob/2822410838fe09647d71fb7b53a8db2136bcbb22/modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/SegmentsExperimentApp.es.js
 to the https://github.com/liferay/liferay-portal/blob/2822410838fe09647d71fb7b53a8db2136bcbb22/modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/App.js

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @beltranrengifo and @claraizquierdo !!

Could you take a look to this PR, we need you adapt the frontend code, so the SegmentExperimentApp.js loads the data form the new endpoint.

Thanks!!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
> Hey @beltranrengifo and @claraizquierdo !!
> 
> Could you take a look to this PR, we need you adapt the frontend code, so the SegmentExperimentApp.js loads the data form the new endpoint.
> 
> Thanks!!

Taking a look!

**Comment by beltranrengifo** [comment]
hi @cgoncas !
I understand that we'll only get `segmentExperimentDataURL` and the rest will be deleted in the final version, right? 
Just to be sure about what to expect to receive in the component. Thxs!

**Comment by cgoncas** [comment]
> hi @cgoncas !
> I understand that we'll only get `segmentExperimentDataURL` and the rest will be deleted in the final version, right?
> Just to be sure about what to expect to receive in the component. Thxs!

Yeahhhh

**Comment by beltranrengifo** [comment]
After a quick analysis, here are my initial thoughts:

As spoken, we'll use this module **Analytics Report Web** as a reference, trying to keep things close between modules.
This module, and the one affected by this task, have this particularities:
- Use a context with a `constants` property, storing everything received initially from the BE.
- Among other required props, like the URL for fetching the data, the FE needs a `isPanelStateOpen` so the fetch only happens in that case. This should be implemented, I believe @cgoncas 
- With this info available, we can mount a [wrapper component just to set up the constants context and handle events](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/App.js), and a [child component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/components/LayoutReports.js) with the main logic that will request conditionally the data based on the panel state.
- I'm not sure if we need everything set up there, like the reducer, two contexts, etc... We'll see during the development. I'll try to keep things as simple as possible.

@claraizquierdo Let me know your thoughts if you please :)

**Comment by beltranrengifo** [comment]
hi @cgoncas !
I see right now we receive in the FE the `segmentExperimentDataURL` property inside another object called `props`
What do you think about get rid of this `props` object, at least for this top-level data?

**Comment by beltranrengifo** [comment]
Hi team!

I’ve just finished making the necessary changes for getting the relevant data from a new endpoint. However I have some really weird behaviours with the variations, something I’m not being able to figure out yet, but what I see is that mixing both worlds (old props with new api endpoint) is messing around because we are forcing some unneeded re-renders that wont exist in the new version, and we are calling the BE more times than needed.

I’ll try to clean the data from BE to FE, and re-evaluate from there…

**Comment by cgoncas** [comment]
> hi @cgoncas ! I see right now we receive in the FE the `segmentExperimentDataURL` property inside another object called `props` What do you think about get rid of this `props` object, at least for this top-level data?

Hey @beltranrengifo !!

I think we better move it to context > segmentExperimentDataURL, as we have it in the other componentes.

Thanks!!

**Comment by beltranrengifo** [comment]
hey @cgoncas 
could you do a quick test just following this steps, whenever you can?

- download new commits on this pr 
- navigate to home page
- create a test
- create a new variation

###**Expected**: 
- the new variation is created with the content from the control variation
- when navigating to the new variation, you can see the content from the control variation
- when editing the new variation, you can add widgets and fragments

###**Actual**
- the variation is created black
- when navigating to the variation, it's blank and the console shows a 404 error
- when editing the variation, no fragments nor widgets can be added to the page

**Comment by beltranrengifo** [comment]
hey @cgoncas 
may I push -f to this branch? I've just rebased and added some minor FE stuff

**Comment by cgoncas** [comment]
> hey @cgoncas may I push -f to this branch? I've just rebased and added some minor FE stuff

Ohhh, ok, I just run ant all, but I can wait for your changes and do it again :)

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

I am receiving the error:

Captura de pantalla 2022-10-04 a las 10 32 18

Do you now how to solve it?

Thanks!!

**Comment by beltranrengifo** [comment]
> Hey @beltranrengifo !!
> 
> I am receiving the error:
> 
> Captura de pantalla 2022-10-04 a las 10 32 18
> 
> Do you now how to solve it?
> 
> Thanks!!

Weird @cgoncas !
that component is now used in the child component. Do you have the last changes?

**Comment by beltranrengifo** [comment]
> > hey @cgoncas may I push -f to this branch? I've just rebased and added some minor FE stuff
> 
> Ohhh, ok, I just run ant all, but I can wait for your changes and do it again :)

The I'll keep changes locally till it's convenient for all to push them, no problem at all

**Comment by beltranrengifo** [comment]
hi @cgoncas !
I think I can solve the `isPanelStateOpen` in FE just by using `getSessionValue, setSessionValue`...
Are you okay with this?

**Comment by cgoncas** [comment]
> > > hey @cgoncas may I push -f to this branch? I've just rebased and added some minor FE stuff
> > 
> > 
> > Ohhh, ok, I just run ant all, but I can wait for your changes and do it again :)
> 
> Then I'll keep changes locally till it's convenient for all to push them, no problem at all

Please, push your changes, maybe it fix everything for me :)

**Comment by beltranrengifo** [comment]
pushed @cgoncas !
I'll have to simplify some stuff, but it should work fine
let me know

**Comment by cgoncas** [comment]
> createSegmentsVariantURL

Hey @beltranrengifo !!

I have pushed a fix with the right URL, could you take a look?

Thanks

**Comment by beltranrengifo** [comment]
> > createSegmentsVariantURL
> 
> Hey @beltranrengifo !!
> 
> I have pushed a fix with the right URL, could you take a look?
> 
> Thanks

Hey @cgoncas !
After some tests and an `ant all` results are the same, variants are no properly created.

`fragmentEntryLinks` are always sent as an empty object on variation creation, we can take a look at that:
image

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

I have updated the PR fixing the problem with the FragmentEntryLinks. Using a Incognito Browser, the info panel load propertly  :) (Yuhuuu).

Could you test that everything is running ok?

Thanks!!

**Comment by beltranrengifo** [comment]
> Hey @beltranrengifo !!
> 
> I have updated the PR fixing the problem with the FragmentEntryLinks. Using a Incognito Browser, the info panel load propertly :) (Yuhuuu).
> 
> Could you test that everything is running ok?
> 
> Thanks!!

Niceeee @cgoncas !
On my way, i'll let you know

**Comment by beltranrengifo** [comment]
hey @cgoncas !
the variant creation seems to be fine 🎉 

now I see another problem:

- create a test
- create a varian
- edit the varian, no matter what
- publish variant changes

**Expected**: the user returns to the page in view mode (control variant is displayed, that's another story)
**Actual**: the user remains in EDIT mode, but in the **control variant**, the blue publish button at the top right constantly redirects the user to the same place when clicked

**Comment by claraizquierdo** [comment]
@beltranrengifo @cgoncas looks like the `editSegmentsVariantLayoutURL` returned by the endpoint is not correct. If I use the old one everything works fine:

- The new one: http://localhost:8080/68e50e93-c7be-fbb3-7e87-b23ebe2463f3?p_l_back_url=&p_l_mode=edit&redirect=http%3A%2F%2Flocalhost%3A8080%2F68e50e93-c7be-fbb3-7e87-b23ebe2463f3%3Fredirect%3Dhttp%253A%252F%252Flocalhost%253A8080%252F68e50e93-c7be-fbb3-7e87-b23ebe2463f3%253Fp_l_back_url%253Dhttp%25253A%25252F%25252Flocalhost%25253A8080%25252Fhome%25253FsegmentsExperienceId%25253D43308%2526p_l_mode%253Dedit%26p_l_back_url%3D%26p_l_mode%3Dedit

- The old one: http://localhost:8080/68e50e93-c7be-fbb3-7e87-b23ebe2463f3?p_l_back_url=http%3A%2F%2Flocalhost%3A8080%2Fhome%3FsegmentsExperienceId%3D43308&p_l_mode=edit&redirect=http%3A%2F%2Flocalhost%3A8080%2F68e50e93-c7be-fbb3-7e87-b23ebe2463f3%3Fp_l_back_url%3Dhttp%253A%252F%252Flocalhost%253A8080%252Fhome%253FsegmentsExperienceId%253D43308%26p_l_mode%3Dedit

**Comment by cgoncas** [comment]
Hey 

> hey @cgoncas ! the variant creation seems to be fine 🎉
> 
> now I see another problem:
> 
> * create a test
> * create a variant
> * edit the variant, no matter what changes
> * publish variant changes
> 
> **Expected**: the user returns to the page in view mode (control variant is displayed, that's another story) **Actual**: the user remains in EDIT mode, but in the **control variant**, the blue publish button at the top right constantly redirects the user to the same place when clicked

Hey @beltranrengifo and @claraizquierdo !!

I think the redirect problem has been fixed, could you take a look?

Thanks!!

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

Could you add the frontend changes to the PR description?

Thanks!!!

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
Hey @CarlosBrichete and @Luizcarlosqueiroz !!

Could you test this PR, it changes the arquitecture of A/B Testing, so we could have some bugs, as the Tango suit is broken right now, could you test it manually?

Thanks!!

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

I think we decided to move the segmentExperimentDataURL from props to context, do you agree with that change, could I do it right now?

Thanks!!

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/portlet/action/GetDataMVCResourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+		);
+	}
+
+	private JSONObject _getPropsJSONObject(
+			HttpServletRequest httpServletRequest, Layout layout, Locale locale,
+			String redirect, long segmentsExperienceId)
+		throws Exception {
+
+		Group group = layout.getGroup();
+
```
Comment (jkappler) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/portlet/action/GetDataMVCResourceCommand.java:399]: @cgoncas, should we use liveGroup here?

Comment (jkappler) [reply modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/portlet/action/GetDataMVCResourceCommand.java:399]: panel

**Comment by Luizcarlosqueiroz** [comment]
Hey @cgoncas! I tested manually with AC docker and AB Test is working well, no changes noticed.

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java [hunk 2]
```diff
@@ -279,6 +286,138 @@ protected void activate(Map<String, Object> properties) {
...
+		throws Exception {
+
+		return HashMapBuilder.<String, Object>put(
+			"context",
+			HashMapBuilder.<String, Object>put(
+				"namespace",
+				_portal.getPortletNamespace(
+					SegmentsPortletKeys.SEGMENTS_EXPERIMENT)
+			).put(
+				"segmentExperimentDataURL",
```
Comment (beltranrengifo) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:299]: Got it @cgoncas !

**Comment by cgoncas** [comment]
ci:reevaluate:1297401_4914

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:reevaluate:1307401_4910

**Comment by beltranrengifo** [comment]
rebased with master & push -f

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Review by beltranrengifo — APPROVED** [review]
I've tested the functionality again, everything seems to work fine.

**Comment by claraizquierdo** [comment]
> Hey
> 
> > hey @cgoncas ! the variant creation seems to be fine 🎉
> > now I see another problem:
> > 
> > * create a test
> > * create a variant
> > * edit the variant, no matter what changes
> > * publish variant changes
> > 
> > **Expected**: the user returns to the page in view mode (control variant is displayed, that's another story) **Actual**: the user remains in EDIT mode, but in the **control variant**, the blue publish button at the top right constantly redirects the user to the same place when clicked
> 
> Hey @beltranrengifo and @claraizquierdo !!
> 
> I think the redirect problem has been fixed, could you take a look?
> 
> Thanks!!

I'm a little bit late but I have tested the code and everything seems to be working fine 👍

**Comment by claraizquierdo** [comment]
ci:reevaluate:1397401_674

**Comment by beltranrengifo** [comment]
ci:reevaluate:1397401_674

**Comment by cgoncas** [comment]
ci:reevaluate:1397401_674

**Comment by jkappler** [comment]
@beltranrengifo, @cgoncas, @claraizquierdo, I will take a look at the session click errors, they shouldn't happen

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Review by claraizquierdo — APPROVED** [review]
LGTM!

**Comment by beltranrengifo** [comment]
ci:forward

**Comment by beltranrengifo** [comment]
ci:reevaluate:1257401_4918

**Comment by beltranrengifo** [comment]
ci:reevaluate:1257401_4918

**Comment by jkappler** [comment]
ci:forward

**Pull request closed without merging**